### PR TITLE
Charlesmchen/yap database connection category

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
PTAL @michaelkirk

**Comment by charlesmchen** [comment]
This method only applies the new category in a few places, but hopefully it demonstrates how easy it will be to migrate more - and then we can get rid of the convenience methods on `TSStorageManager`.

File: SignalServiceKit/src/Storage/YapDatabaseConnection+OWS.m [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+//
+
+#import "YapDatabaseConnection+OWS.h"
+#import <YapDatabase/YapDatabaseTransaction.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation YapDatabaseConnection (OWS)
+
+- (id)objectForKey:(NSString *)key inCollection:(NSString *)collection
```
Comment (michaelkirk) [inline SignalServiceKit/src/Storage/YapDatabaseConnection+OWS.m]: nit: `(nullable id)`

Comment (charlesmchen) [inline SignalServiceKit/src/Storage/YapDatabaseConnection+OWS.m]: Done. In fact, I went you one better and significantly cleaned up these methods - DRY'ing up code, adding asserts, etc.

**Pull request merged**